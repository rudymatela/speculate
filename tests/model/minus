max expr size  =    5
max  #-tests   =  500
min  #-tests   =   25  (to consider p ==> q true)
max  #-vars    =    2  (for inequational and conditional laws)

_ :: Int  (holes: Int)
0 :: Int
1 :: Int
id :: Int -> Int
negate :: Int -> Int
(+) :: Int -> Int -> Int
(-) :: Int -> Int -> Int

             id x == x
            x - y == x + negate y
negate (negate x) == x
            x + 0 == x
            x + y == y + x
     negate x + y == negate (x + negate y)
     x + negate x == 0
      (x + y) + z == x + (y + z)

             x <= x + 1
         x - 1 <= x
negate (x + 1) <= negate x

