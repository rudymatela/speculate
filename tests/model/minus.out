max expr size  =    5
  |- on ineqs  =    4
  |- on conds  =    4
max  #-tests   =  500
min  #-tests   =   25  (to consider p ==> q true)
max  #-vars    =    2  (for inequational and conditional laws)

_ :: Int  (holes: Int)
0 :: Int
1 :: Int
id :: Int -> Int
(+) :: Int -> Int -> Int
negate :: Int -> Int
(-) :: Int -> Int -> Int

             id x == x
negate (negate x) == x
            x + 0 == x
     x + negate x == 0
            x + y == y + x
            x - y == x + negate y
   negate (x + y) == negate x + negate y
      (x + y) + z == x + (y + z)

    x <= x + 1
x - 1 <= x

